modules/basegl/glsl/img_channel_combine.frag

```diff
@@ -0,0 +1,54 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2013-2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ *********************************************************************************/
+
+#include "utils/structs.glsl"
+
+uniform ImageParameters outportParameters;
+
+uniform sampler2D inport0Color;
+uniform sampler2D inport1Color;
+uniform sampler2D inport2Color;
+uniform sampler2D inport3Color;
+
+uniform int redChannel;
+uniform int greenChannel;
+uniform int blueChannel;
+uniform int alphaChannel;
+
+uniform bool use_alpha_texture = true;
+uniform float alpha = 1.0;
+
+void main() {
+    vec2 texCoords = gl_FragCoord.xy * outportParameters.reciprocalDimensions;
+    float r = texture(inport0Color, texCoords)[redChannel];
+    float g = texture(inport1Color, texCoords)[greenChannel];
+    float b = texture(inport2Color, texCoords)[blueChannel];
+    float a = mix(alpha,texture(inport3Color, texCoords)[alphaChannel], use_alpha_texture);
```

Comment:
mix with float/bool is not supported on <4.0 glsl version
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/mix.xhtml
Convert the bool to float

Reply:
Supported according to standard in 3.3 (https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.3.30.pdf) 
![image](https://user-images.githubusercontent.com/534670/60249433-22b86580-98c5-11e9-913b-6bebc32603d4.png)

Reply:
Interesting. Seems like they differ. Then we go with the spec. :)

Reply:
think the difference might be if the bool is a bool or a bvec. the case with just  a bool and not a bvec might not work on 3.3...

Reply:
> 
> 
> think the difference might be if the bool is a bool or a bvec. the case with just a bool and not a bvec might not work on 3.3...

They are all scalars so it shouldn't be a problem